widget/tabcontainer.go

```diff
@@ -20,6 +20,8 @@ type TabContainer struct {
 	OnChanged   func(tab *TabItem)
 	current     int
 	tabLocation TabLocation
+	focused     bool
+	renderer    tabContainerRenderer
```

Comment:
This is a no-no - a widget must not retain a reference to it's current renderer

Reply:
This is one of the difficult, deep issues. I could not find any other way to do it.

Reply:
From the code that uses it it's hard to see the intent, was it to change buttons or highlight that one had been pressed? if it's the latter then maybe tabButton is the place for that code? otherwise the keyboard handling code can change selected item that the render then updates to reflect.

Reply:
Actualy, on the toolbar, this functionality is not needed and has been removed. But on the Toolbar, there is a similar use of the renderer.
The problem is that the toolbar must update the state of the boolbar buttons, as it receives key-presses. But it has no reference to its buttons. To avoid the renderer reference, I have instead added a buttons array in the toolbar.

Reply:
Oops. Is there a way to force detachement of all renderers, so this can be tested?

Reply:
Currently we do not have that test capability, but it’s a good suggestion.